integration/build_test.go

```diff
@@ -179,3 +179,39 @@ func failNowIfError(t *testing.T, err error) {
 		t.Fatal(err)
 	}
 }
+
+// TestExpectedBuildFailures verifies that `skaffold build` fails in expected ways
+func TestExpectedBuildFailures(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	tests := []struct {
+		description string
+		dir         string
+		filename    string
+		args        []string
+		gcpOnly     bool
+	}{
+		{
+			description: "jib is too old",
+			dir:         "testdata/jib",
+			args:        []string{"-p", "old-jib"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			if test.gcpOnly && !ShouldRunGCPOnlyTests() {
+				t.Skip("skipping gcp only test")
+			}
+			if !test.gcpOnly && ShouldRunGCPOnlyTests() {
+				t.Skip("skipping test that is not gcp only")
+			}
+
+			err := skaffold.Build(test.args...).WithConfig(test.filename).InDir(test.dir).Run(t)
```

Comment:
I think `.RunOrFail` does that

Reply:
RunOrFail fails the test on error, whereas the error is a sign of success here.

Reply:
I updated this test to check for expected error text as it would spuriously pass if (say) the wrong skaffold version was picked up (_"Your Skaffold version might be too old"_).

Reply:
oh, right!